File: apk/src/main/java/uk/co/alt236/apk/util/Hasher.java
package uk.co.alt236.apk.util;

import org.apache.commons.codec.digest.DigestUtils;

import java.io.File;
import java.io.FileInputStream;
import java.io.IOException;
import java.io.InputStream;

public class Hasher {

    public String md5Hex(final byte[] input) {
        return DigestUtils.md5Hex(input);
    }

    public String md5Hex(final File input) {
        String retVal;
        try {
            FileInputStream fis = new FileInputStream(input);
            retVal = DigestUtils.md5Hex(fis);
            fis.close();
        } catch (IOException e) {
            throw new IllegalStateException(e.getMessage(), e);
        }

        return retVal;
    }

    public String sha1Hex(final byte[] input) {
        return DigestUtils.sha1Hex(input);
    }

    public String sha1Hex(final File input) {
        String retVal;
        try {
            FileInputStream fis = new FileInputStream(input);
            retVal = DigestUtils.sha1Hex(fis);
            fis.close();
        } catch (IOException e) {
            throw new IllegalStateException(e.getMessage(), e);
        }

        return retVal;
    }

    public String sha256Hex(final byte[] input) {
        return DigestUtils.sha256Hex(input);
    }

    public String sha256Hex(final File input) {
        String retVal;
        try {
            FileInputStream fis = new FileInputStream(input);
            retVal = DigestUtils.sha256Hex(fis);
            fis.close();
        } catch (IOException e) {
            throw new IllegalStateException(e.getMessage(), e);
        }

        return retVal;
    }

    public String sha256Hex(final InputStream inputStream) {
        String retVal;
        try {
            retVal = DigestUtils.sha256Hex(inputStream);
            inputStream.close();
        } catch (IOException e) {
            throw new IllegalStateException(e.getMessage(), e);
        }

        return retVal;
    }

}

File: apk/src/main/java/uk/co/alt236/apk/util/Hasher.kt
package uk.co.alt236.apk.util

import org.apache.commons.codec.digest.DigestUtils
import java.io.File
import java.io.FileInputStream
import java.io.InputStream

class Hasher {

    fun md5Hex(input: ByteArray): String {
        return DigestUtils.md5Hex(input)
    }

    fun md5Hex(input: File): String {
        val retVal: String
        val fis = FileInputStream(input)
        retVal = DigestUtils.md5Hex(fis)
        fis.close()

        return retVal
    }

    fun sha1Hex(input: ByteArray): String {
        return DigestUtils.sha1Hex(input)
    }

    fun sha1Hex(input: File): String {
        val retVal: String
        val fis = FileInputStream(input)
        retVal = DigestUtils.sha1Hex(fis)
        fis.close()
        return retVal
    }

    fun sha256Hex(input: String): String {
        return DigestUtils.sha256Hex(input)
    }

    fun sha256Hex(input: ByteArray): String {
        return DigestUtils.sha256Hex(input)
    }

    fun sha256Hex(input: File): String {
        val retVal: String
        val fis = FileInputStream(input)
        retVal = DigestUtils.sha256Hex(fis)
        fis.close()
        return retVal
    }

    fun sha256Hex(inputStream: InputStream): String {
        val retVal = DigestUtils.sha256Hex(inputStream)
        inputStream.close()
        return retVal
    }

}
